### DISCUSSION TITLE
Is it possible to use friendly_name instead of device_name for MQTT power monitoring?

### DISCUSSION THREAD

**Discussion by AleXSR700** [arendst/Tasmota#16490] [Q&A]
Hello everyone,
sorry for the confusing title. Not sure how to name it.

I have multiple Shelly 2.5 and as you know the monitor the power for both outputs. And both are shown separately in the WebUI. However, when using MQTT, despite setting two friendly_names, the power monitoring data is always sent as device_name with a number suffix.
I would prefer if the information were sent using the friendly_name 1 & 2. This would make more sense because that uniquely identifies the consumer whereas the device_name does not.

Is this somehow possible?

Thank you
Alex

**Comment by sfromis** [comment]
Neither device name nor friendly name is or can be used in the MQTT topic, it is up to you to configure the `Topic` you want, instead of the default one. You do not get separate topics per power channel.

**Reply by AleXSR700** [reply]
This is true, but the two sensors are marked with a suffix number. Is this done internally in Home Assistant?

And can I assign a topic per channel?
It should simply be that channel one is friendly_name1 and channel 2 is friendly_name2.

**Reply by sfromis** [reply]
I already mentioned that you do not get separate topics per channel, and as friendly names are designed as a human-readable word sequences, they are not suitable for, or possible to directly use as topic. What you have is the possibility to change topic for the whole device, and nothing more in config.

The payload for an energy monitoring device with two channels is that they are index 1 and 2, and this is an architecture not amendable to customization in settings, and is likely what HA is picking up on. I'm not using HA, but my expectation would be that it may be able to coordinate the friendly names with the channel indexes, but this is really outside of what I know about.

**Reply by AleXSR700** [reply]
Hmmm.
The thing is, and again I do not know if this is Home Assitant or Tasmota, that when I look at the matching switches, the switches are labelled with friendly_name and not numbered device_names.

So the two switches are using friendly_names but the matching power monitoring uses device_names.

**Reply by barbudor** [reply]
Are u using HaTasmota plug-in?
If yes this the one who make HA display what it display
So you may escalade that to HaTasmota and if anything is missing in Tasmota Discovery it could be added

**Reply by sfromis** [reply]
Well, I've made sure not to know much about HA, so I have no answers there.... However, Tasmota itself is not using the friendly names for anything, except for providing them to external platforms like HA or Amazon Alexa. That means that whatever happens (or not) with friendly name is up to that platform.

There is nowhere in the individual messages about neither switches nor energy consumption that the friendly names are used in any way, but they are listed in the discovery topics used by features like Tasmota integration in HA. This allows that platform to use the friendly names where they fit, and I suppose that with an energy monitoring device, it would make sense to apply them to the energy consumption data as well as to the relays.

**Reply by AleXSR700** [reply]
Yes, the Tasmota integration is being used. But that is automatic. The system recognizes the device as a Tasmota device. I have raised the issue there and will report back here if it can be solved in HA integration :)
